rouskinhf/list_datapoints.py

```diff
@@ -208,27 +227,29 @@ def drop_duplicates(df:pd.DataFrame, **kwargs):
             n_duplicates_datapoints = drop_duplicates(df, subset=['sequence','paired_bases'], inplace=True, ignore_index=True)
 
         # Keep only one datapoint per sequence and dms
-        if 'dms' in df.columns:
-            if not 'paired_bases' in df.columns: n_duplicates_datapoints = 0
-            n_duplicates_datapoints += drop_duplicates(df, subset=['sequence','dms'], inplace=True, ignore_index=True)
-        
+        for signal in ['dms', 'shape']:
+            if signal in df.columns:
+                if not 'paired_bases' in df.columns: n_duplicates_datapoints = 0
+                n_duplicates_datapoints += drop_duplicates(df, subset=['sequence', signal], inplace=True, ignore_index=True)
+            
         # If there are multiple structures / dms with the same structure, keep none
-        n_same_seq_datapoints = drop_duplicates(df, subset=['sequence'], inplace=True, ignore_index=True, keep='first' if not ('paired_bases' in df.columns or 'dms' in df.columns) else False)
+        n_same_seq_datapoints = drop_duplicates(df, subset=['sequence'], inplace=True, ignore_index=True, keep='first' if not ('paired_bases' in df.columns or 'dms' in df.columns or 'shape' in df.columns) else False)
```

Comment:
I don't understand how this removes all datapoints if the structures/dms/shape are different